omod-2.2/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_2/ProgramAttributeTypeResource2_2.java

```diff
@@ -39,31 +46,47 @@ public DelegatingResourceDescription getRepresentationDescription(Representation
 		}
 		return super.getRepresentationDescription(rep);
 	}
-	
+
 	@Override
 	public ProgramAttributeType getByUniqueId(String uuid) {
 		return Context.getProgramWorkflowService().getProgramAttributeTypeByUuid(uuid);
 	}
-	
+
 	@Override
 	public ProgramAttributeType newDelegate() {
 		return new ProgramAttributeType();
 	}
-	
+
 	@Override
 	public ProgramAttributeType save(ProgramAttributeType programAttributeType) {
 		return Context.getProgramWorkflowService().saveProgramAttributeType(programAttributeType);
 	}
-	
+
 	@Override
 	public void purge(ProgramAttributeType programAttributeType, RequestContext requestContext) throws ResponseException {
 		Context.getProgramWorkflowService().purgeProgramAttributeType(programAttributeType);
 	}
-	
+
 	@Override
 	protected NeedsPaging<ProgramAttributeType> doGetAll(RequestContext context) throws ResponseException {
 		return new NeedsPaging<ProgramAttributeType>(Context.getProgramWorkflowService()
 		        .getAllProgramAttributeTypes(),
 		        context);
 	}
+
+	@PropertyGetter("concept")
+	public Object getConcept(ProgramAttributeType delegate) {
+		if (!OpenmrsUtil.nullSafeEquals(delegate.getDatatypeClassname(), ConceptDatatype.class.getCanonicalName()))
+			return null;
+		else {
+			Concept concept;
+			String id = delegate.getDatatypeConfig();
+			if (Pattern.compile("\\d+").matcher(id).matches()) {
```

Comment:
Is this code written to check if id is a valid Integer? If so, then there might already be a utils method for this, which is more performant and clearer than regex. Also some related reading: https://www.baeldung.com/java-check-string-number (provides alternatives on Apache Commons library for this purpose). Might make for clearer code.

Reply:
@gsluthra Thought of having this in a util file but i don't see any generic util file in the webservices.rest module. The other option could be separating the if condition into a private method in the same class. Please suggest what can be the best option

Reply:
How about the Apache Commons library?

Reply:
@binduak what do you think of the above comment?

Reply:
@dkayiwa handled the feedback changes as per the suggestions